multiplexer/appd/run.go

```diff
@@ -82,50 +76,56 @@ func (a *Appd) Start(args ...string) error {
 	if err := cmd.Start(); err != nil {
 		return fmt.Errorf("failed to start %s: %w", a.path, err)
 	}
+	a.cmd = cmd
+	return nil
+}
 
-	a.pid = cmd.Process.Pid
+func (a *Appd) IsRunning() bool {
+	return !a.IsStopped()
+}
 
-	go func() {
-		// This waits whether process exits naturally or is killed by Stop()
-		if err := cmd.Wait(); err != nil {
-			log.Printf("Process finished with error: %v\n", err)
-		} else {
-			log.Printf("Process finished with no error\n")
-		}
-		a.pid = AppdStopped // Always reset PID when process ends
-	}()
+func (a *Appd) IsStopped() bool {
+	// Never started or failed to start
+	if a.cmd == nil || a.cmd.Process == nil {
+		return true
+	}
 
-	return nil
+	// If ProcessState is not nil, it means Wait() was called and the process has finished
+	// (either by exiting normally or being terminated by a signal)
+	if a.cmd.ProcessState != nil {
+		return true
+	}
+
+	// ProcessState is nil, which means the process is still running
+	return false
 }
 
-// Stop terminates the running appd process if it exists.
+// Stop terminates the running appd process if it exists and waits for it to fully exit.
 func (a *Appd) Stop() error {
-	if a.pid == AppdStopped {
-		return nil
+	if a.cmd == nil {
+		return fmt.Errorf("can not stop appd because cmd is nil")
```

Comment:
the previous behaviour was not to error if `IsStopped` was true i.e. if you call Stop on an already stopped multiplexer so we should be careful this doesn't break anything.